source/web-hosting/applications.rst

```diff
@@ -81,6 +81,31 @@ To install dependencies::
 
 Then see ``nodejs/app.js`` for a minimum base application.
 
+Forwarding requests to your application server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The SRCF uses Apache to serve websites so if you need to forward requests to a backend web app, for example a Django, Rails or Express server, then you will need to forward web requests. Make sure you run your application server on ``sinkhole`` rather than ``pip`` or another SRCF server.
+
+Using TCP ports
+^^^^^^^^^^^^^^^
+
+You will need to pick a port (we've used 999 here but you should pick a different one above 1024) and configure your application to bind to that port. Then stick the following in you ``.htaccess`` file::
+
+    RequestHeader set X-Forwarded-Proto expr=%{REQUEST_SCHEME}
+    RequestHeader set Host expr=%{HTTP_HOST}
+    RewriteCond %{HTTP_HOST} ^your\.hostname\.tld$
```

Comment:
fwiw the RequestHeader and RewriteConds were specific for my particular application; you don't need it for the proxying in general.

For forwarding to a _port_ it's actually risky to trust X-Forwarded-Proto and Host headers, since anyone logged into sinkhole could fake them.  Maybe it's actually worth putting a security warning note here about the risks of proxying over TCP?

Reply:
Yeah that's true - I'd forgotten about the security issues of blindly trusting those headers over TCP. I will add a note about it and advise that people attempt to use sockets if possible.

RE: headers: I tend to use something like this in my nginx proxy configs -- is it worth Apache-ifying and going in to details about what they mean?

```nginx
proxy_set_header Host              $http_host;
proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Real-IP         $remote_addr;
proxy_set_header Upgrade           $http_upgrade;
```